cvmfs/publish/repository_managed.cc

```diff
@@ -83,6 +83,7 @@ void Publisher::ManagedNode::ClearScratch() {
 
 
 int Publisher::ManagedNode::Check(bool is_quiet) {
+  ServerLockFileCheck publish_check(publisher_->is_publishing_);
```

Comment:
Also here, we are taking the lock for a short time during the check, aren't we?

Reply:
Yes.  This is necessary when using a standard POSIX lock file: the only way to tell if something else holds the lock is to attempt to acquire the lock.

In general, this is correct behaviour for locking semantics: any code that does a test for "is this lock held?" rather than "can I acquire this lock?" is immediately suspicious since it suggests a race condition.